### PR TITLE
Keep track of the amplitude properties with the identify function.

### PR DESCRIPTION
**Pull request by aikewoody**
# Description

These changes are made in response to the review from the initial PR: https://github.com/zoe/covid-tracker-react-native/pull/1528

## On what platform have you tested the change?

- [ ] Android - Emulator
- [ ] Android - Physical device
- [ ] iOS - Emulator
- [ ] iOS - Physical device

## Screenshots

[comment]: <> (Please attach screenshots showing the change, if relevant)

## Checklist (for submission)

- [ ] Analytics/tracking events have been added (if relevant)
- [ ] Added new screen(s) to `src/features/screens.ts` and new Redux slice(s) to `mocks/mockedInitialState.ts` (if relevant)

## Checklist (for reviewers)

- [ ] Checked against Figma designs (if relevant)
- [ ] Checked that data has been successfully saved to the backend (if relevant)

## Out of scope and potential follow-up

[comment]: <> (Are there any related changes that you plan to do in a follow-up PR or voluntarily excluded from the scope?)

### CODE REVIEW

File: src/core/Analytics.ts [hunk 1]
```diff
@@ -1,15 +1,14 @@
...
 import Constants from '@covid/utils/Constants';
 import * as Amplitude from 'expo-analytics-amplitude';
 
 let isInitialized = false;
 
-type TAdditionalProps = {
+type TUserProperties = {
   isTester?: boolean;
-  Experiment_001?: string;
-  Experiment_mhip?: string;
```
Comment (selvach) [inline src/core/Analytics.ts:12]: Can we keep `Experiment_mhip` as the original name users will have already been tagged with please. This should be either cohort a, b or c. 

Maybe change `experimentMHPGroup` -> `Experiment_mhip_2` (or similar?)

File: src/core/Analytics.ts [hunk 2]
```diff
@@ -121,34 +120,30 @@ function initialize(): void {
...
-  track(events.VIEW_SCREEN_MHP, { mh_insight_cohort, screenName, testGroupId });
-}
-
 export function trackScreenView(screenName: string): void {
   track(events.VIEW_SCREEN, { screenName });
 }
 
-export function identify(additionalProps?: TAdditionalProps): void {
+export function trackModalView(modalName: string): void {
+  track(events.VIEW_MODAL, { modalName });
```
Comment (selvach) [inline src/core/Analytics.ts:138]: Nice

File: src/components/Modal.tsx [hunk 3]
```diff
@@ -1,16 +1,23 @@
...
 import { Modal as RNModal, ScrollView, StyleSheet } from 'react-native';
 
 interface IProps {
   children?: React.ReactNode;
+  modalName?: string;
   onRequestClose: () => void;
   visible: boolean;
 }
 
 export default function Modal(props: IProps) {
```
Comment (selvach) [inline src/components/Modal.tsx:13]: Nice!

File: src/features/AppCoordinator.ts [hunk 4]
```diff
@@ -143,7 +144,15 @@ export class AppCoordinator extends Coordinator implements ISelectProfile, IEdit
       this.goToVersionUpdateModal();
     }
 
-    Analytics.identify();
+    if (startupInfo?.mh_insight_cohort) {
```
Comment (selvach) [inline src/features/AppCoordinator.ts:147]: Nice!

**Review by selvach — COMMENTED** [review]
I think this is an improvement. The A/B Test & analytics code is more localised and less spread across the repo.

File: src/features/AppCoordinator.ts [hunk 5]
```diff
@@ -119,13 +120,13 @@ export class AppCoordinator extends Coordinator implements ISelectProfile, IEdit
 
   async init(setUsername: (username: string) => void, setPatients: (patients: string[]) => void) {
     let user: TUserResponse | null = null;
-    let patientId: string | null = null;
```
Comment (selvach) [inline src/features/AppCoordinator.ts:122]: This is a small change, which I hope is fine... but who knows without tests :) 

I would leave it out.

Comment (aikewoody) [reply src/features/AppCoordinator.ts:122]: You're right, I couldn't resist my refactor urge but better be safe than sorry!

**Review by selvach — APPROVED** [review]
Nice!

**Pull request merged**